### PR TITLE
Add Stable Diffusion Interpolation Example

### PR DESCRIPTION
**Pull request by nateraw**
Adds `interpolate_stable_diffusion.py`, which is a pipeline that lets you generate images as you interpolate between different prompts/seeds.

Its `__call__` fn is the same as `StableDiffusionPipeline`, but with the added `text_embeddings` kwarg. The `walk` function is where the logic happens. Fine with renaming these if need be so that `walk` becomes `__call__`. 

Here's how I have been using it (from within `examples/community` dir)

```python
from interpolate_stable_diffusion import StableDiffusionWalkPipeline
import torch

pipe = StableDiffusionWalkPipeline.from_pretrained(
    "CompVis/stable-diffusion-v1-4",
    revision='fp16',
    torch_dtype=torch.float16,
    safety_checker=None,  # Very important for videos...lots of false positives while interpolating
).to('cuda')

frame_filepaths = pipe.walk(
    prompts=['a dog', 'a cat', 'a horse'],
    seeds=[42, 1337, 1234],
    num_interpolation_steps=16,
    output_dir='./dreams',
    batch_size=24,
    height=512,
    width=512,
    guidance_scale=8.5,
    num_inference_steps=50,
)
```

Though it seems (I think) that once it's merged, you'll be able to use `community='interpolate_stable_diffusion'` in the `from_pretrained` fn instead of having to be in the `examples/community` dir.

### CODE REVIEW

File: examples/community/README.md [hunk 1]
```diff
@@ -5,4 +5,6 @@
 | Example   |      Description      |      Author      |    Colab  |
 |:----------|:----------------------|:-----------------|----------:|
 | CLIP Guided Stable Diffusion | Doing CLIP guidance for text to image generation with Stable Diffusion| [Suraj Patil](https://github.com/patil-suraj/) | [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/huggingface/notebooks/blob/main/diffusers/CLIP_Guided_Stable_diffusion_with_diffusers.ipynb) |
-| One Step U-Net (Dummy) | [Patrick von Platen](https://github.com/patrickvonplaten/) | - |
+| One Step U-Net (Dummy) | - | [Patrick von Platen](https://github.com/patrickvonplaten/) | - |
```
Comment (nateraw) [inline examples/community/README.md:8]: btw fixed this table here... @patrickvonplaten your name was under the `Description` column.

File: examples/community/README.md [hunk 2]
```diff
@@ -5,4 +5,6 @@
...
+| Stable Diffusion Interpolation | Interpolate the latent space of Stable Diffusion between different prompts/seeds | [Nate Raw](https://github.com/nateraw/) | - |
```
Comment (nateraw) [inline examples/community/README.md:9]: will add notebook once this is merged

File: examples/community/interpolate_stable_diffusion.py [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+        Disable sliced attention computation. If `enable_attention_slicing` was previously invoked, this method will go
+        back to computing attention in one step.
+        """
+        # set slice_size = `None` to disable `attention slicing`
+        self.enable_attention_slicing(None)
+
+    @torch.no_grad()
+    def __call__(
+        self,
+        prompt: Optional[Union[str, List[str]]] = None,
```
Comment (nateraw) [inline examples/community/interpolate_stable_diffusion.py:153]: prompt becomes optional, as you can either pass that or `text_embeddings`

File: examples/community/interpolate_stable_diffusion.py [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+        width: Optional[int] = 512,
+        guidance_scale: Optional[float] = 7.5,
+        num_inference_steps: Optional[int] = 50,
+        eta: Optional[float] = 0.0,
+    ) -> List[str]:
+        """
+        Walks through a series of prompts and seeds, interpolating between them and saving the results to disk.
+
+        Args:
+            prompt (`List[str]`):
```
Comment (nateraw) [inline examples/community/interpolate_stable_diffusion.py:443]:
```suggestion
            prompts (`List[str]`):
```

**Review by nateraw — COMMENTED** [review]
Few notes

**Comment by patrickvonplaten** [comment]
Thanks a lot @nateraw !

**Pull request merged**